sklearn/tests/test_cross_validation.py

```diff
@@ -438,6 +438,109 @@ def test_label_kfold():
     assert_raises(ValueError, cval.LabelKFold, labels, n_folds=3)
 
 
+def test_binnedstratifiedkfold_balance():
+    for _ in range(10):
+        n_folds = 2 + int(10*np.random.rand())
+        y = np.random.randn(30)
+        np.random.shuffle(y)
+        sizes = []
+
+        bskf = cval.BinnedStratifiedKFold(y, n_folds=n_folds,
+                                          shuffle = False, random_state=None)
+
+        bins = np.array([np.percentile(y, q) for q in range(n_folds)])
+        for  train_index, test_index in bskf:
+            sizes.append(
+                        len(test_index)
+                        )
+        assert_true((np.max(sizes) - np.min(sizes)) <= 1)
+        assert_equal(np.sum(sizes), bskf.n)
+
+
+def test_binnedstratifiedkfold_bin_spacing():
+    "check if the binned `y` falls into bins of equal size (+/- 1)"
+    for _ in range(10):
+        n_folds = 2 + int(10*np.random.rand())
+        y = np.random.randn(30)
+        np.random.shuffle(y)
+
+        skf = cval.BinnedStratifiedKFold(y, n_folds=n_folds,
+                                         shuffle = False, random_state=None)
+
+        #bins = np.percentile(y, np.arange(n_folds))
+        bins = np.array([np.percentile(y, q) for q in range(n_folds)])
+
+        for  train_index, test_index in skf:
+            y_test = y[test_index]
+            hist_test, _ = np.histogram( y_test, bins = bins )
+            assert_true(all(abs(hist_test - np.mean(hist_test)) <= 1),
+                        msg = "y_test falls into bins of too ragged sizes")
+
+            y_train = y[train_index]
+            hist_train, _ = np.histogram( y_test, bins = bins )
+            assert_true(all(abs(hist_train - np.mean(hist_train)) <= 1),
+                        msg = "y_train falls into bins of too ragged sizes")
+
+
+def test_binnedstratifiedkfold_has_more_stable_distribution_moments_between_folds():
```

Comment:
I would just leave it as it is, though I don't mind if you amend it.

On Wed, Apr 13, 2016 at 9:32 AM, Raghav R V <notifications@github.com>
wrote:

> In sklearn/tests/test_cross_validation.py
> <https://github.com/scikit-learn/scikit-learn/pull/6598#discussion_r59580505>
> :
>
> > +        #bins = np.percentile(y, np.arange(n_folds))
> > +        bins = np.array([np.percentile(y, q) for q in range(n_folds)])
> > +
> > +        for  train_index, test_index in skf:
> > +            y_test = y[test_index]
> > +            hist_test, _ = np.histogram( y_test, bins = bins )
> > +            assert_true(all(abs(hist_test - np.mean(hist_test)) <= 1),
> > +                        msg = "y_test falls into bins of too ragged sizes")
> > +
> > +            y_train = y[train_index]
> > +            hist_train, _ = np.histogram( y_test, bins = bins )
> > +            assert_true(all(abs(hist_train - np.mean(hist_train)) <= 1),
> > +                        msg = "y_train falls into bins of too ragged sizes")
> > +
> > +
> > +def test_binnedstratifiedkfold_has_more_stable_distribution_moments_between_folds():
>
> I was thinking more like you could name it simply to
> test_binned_stratified_kfold_stable_dist_moments and add a comment inside
> the test explaining it in more detail if you prefer?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/6598/files/c0af740545c373a3dc5f6a95c415c96873e56491#r59580505>
>